### ISSUE TITLE
Network activity towards RFC1918 (private network ranges)

### ISSUE THREAD

**Issue by RSMCx1** [prysmaticlabs/prysm#8459]
Hello,

I've recently reviewed the firewall log of my Ethereum node, and I see allot of (blocked) outgoing traffic to private network ranges (RFC 1918) not belonging to my internal network.

To name a few examples:
172.17.0.1:9000 TCP
172.31.16.80:9000 TCP
172.17.0.1:9000 TCP
192.168.1.222:13000 TCP
10.36.8.30:5000 TCP
10.36.7.5:5000 TCP

I've found an issue in the past that addresses the observed behavior: https://github.com/prysmaticlabs/prysm/issues/5663

It was mentioned the issue would most likely be resolved with the discovery 5 protocol, i'm uncertain what discovery version is currently being used. What I do know is that my node is currently running Prysm v1.2.1.

I don't think it can do a lot of harm, but it is certainly not something that should be allowed to happen. I'm also concerned about the possible implications of, for instance, nodes advertising the same private IP as another one.

Is this being addressed or is there a way to ignore nodes that broadcast private IP addresses?

Additionally, can anybody that broadcasts a private IP be considered malicious? Or is a configuration mistake the result of the behavior that is being witnessed? 

Thanks!

**Comment by nisdas** [comment]
@RSMCx1 can you run the node with this flag ?

```
--p2p-blacklist 10.0.0.0/8,172.16.0.0/12,192.168.0.0/16,100.64.0.0/10,169.254.0.0/16
```
This should ban all private IPs, so in that case you will not dial them.

**Comment by nisdas** [comment]
Also related #8466

**Comment by RSMCx1** [comment]
@nisdas I agree it's a tricky issue. Top of mind I cannot think of a very easy solution, at some point it should be addressed however. This behavior should not exist on a public network, a node is no use anyways if the other nodes do not know how to connect to it.

**Comment by nisdas** [comment]
@RSMCx1 If you do run into this again, you can use this flag
```
--p2p-denylist=private
```
This was added in #8648 . On disabling private dials, that is tricky as it will prevent us from launching small scale local testnets and also peering with any local nodes that we do have. Peers can occasionally be misconfigured or simply provide us with bogus enrs. We have very little control over this, which is why I will be closing this issue unless you feel that it should still stay open.

**Comment by dtto** [comment]
I stumbled upon this issue when trying to figure out why prysm was sending traffic to RFC1918 addresses. FWIW, I am not sure why this is the default behavior. For use cases where traffic needs to be sent to RFC1918 it could be enabled.

Also, for anyone else who wants to disable it and uses a yaml config file, it may be done using:

`p2p-denylist: ["10.0.0.0/8","172.16.0.0/12","192.168.0.0/16","100.64.0.0/10","169.254.0.0/16"]`

When I tried simply adding `p2p-denylist: private' to the yaml config, the client threw `level=error msg="Mismatched type for flag 'p2p-denylist'. Expected '[]interface{}' but actual is 'string'"`.
